services/submission/src/main/java/app/coronawarn/server/services/submission/config/SubmissionServiceConfig.java

```diff
@@ -0,0 +1,64 @@
+package app.coronawarn.server.services.submission.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationProperties(prefix = "services.submission")
+public class SubmissionServiceConfig {
+
+  // Exponential moving average of the last N real request durations (in ms), where
+  // N = fakeDelayMovingAverageSamples.
+  private Double initialFakeDelayMilliseconds;
+  private Double fakeDelayMovingAverageSamples;
+  private Integer retentionDays;
```

Comment:
I would suggest to use java.time.Duration instead of int. Spring Boot should convert the input automatically. Using Duration in DiagnosisKey would make the implementation easier. Something like
`return Duration.between(Instant.ofEpochMilli(rollingStartNumber), Instant.now())
      .compareTo(durationToRetain) <= 0;`

Reply:
Hey @elch78,
thanks for the feedback! Yes I guess you are right and we should use domain specific types where ever possible. However, please understand that the purpose of this PR is to "only" remove the `@Value` annotations and not to much more refactoring above that. Since this PR is already clocking in at almost 1000 LOC delta, I kindly ask you to create an issue for this so we can pick it up in a separate (more manageable) PR. Thanks! :)

Reply:
Agree